File: Codes/Matlab/Chapters/Chapter3/Logistic_example.m
%basic Scalar ODE 
clear
close all

%define parameters
params=[.1,100];
%Define the time interval
t_start = 0;
t_stop = 4;
%Initial Condition
IC=.2;
%Solving the ODE
[t,y]=ode45(@(t,Y) rhs(t,Y,params), [t_start t_stop], IC); 
%plotting the ODE
plot(t,y,LineWidth=2)
xlabel('Time')
ylabel('Y')
%Define the right-hand-side function
function f=rhs(t,Y,params)
r=params(1);
k=params(2);
y=Y;
f=r*y*(k-y);
end

File: Codes/Python/Chapters/Chapter3/Logistic_example.py

"""
Created on Wed Dec 22 21:11:32 2021 Basic ODE Script
@author: cogan
"""
from IPython import get_ipython 
get_ipython().magic('reset -sf')
import numpy as np
from matplotlib import pyplot
# import plotting library
from scipy.integrate import solve_ivp
#Define the right-hand-side of the differential equation
pyplot.close('all')
pyplot.style.use('tableau-colorblind10')  #Fixes colorscheme to be accessible

def rhs(t,Y,params): 
    r, K = params 
    y=Y
    return [r*y*(K-y)]
 # Initial Condition
IC=.2
#Intial time
tstart=0
#Final time
tstop=4
tspan = np.linspace(tstart, tstop, 100) 
#Parameters
params=[.1,100]
y_solution = solve_ivp(lambda t,Y: rhs(t,Y,params), [tstart,tstop], [IC], method='RK45',t_eval=tspan)
pyplot.plot(y_solution.t, y_solution.y[0],'k',label='y(t)') 
pyplot.ylabel('y(t) ',fontsize=16) 
pyplot.xlabel('t',fontsize=16) 
#pyplot.savefig('Logistic.png')
